File: backend/src/Exceptions/StreamingException.php
<?php

declare(strict_types=1);

namespace Quantis\AIPortfolioAssistant\Exceptions;

/**
 * Exception for SSE streaming errors
 */
class StreamingException extends AIAssistantException
{
    /**
     * Create exception for connection failure
     */
    public static function connectionFailed(string $reason): self
    {
        return new self("SSE connection failed: {$reason}", 503);
    }

    /**
     * Create exception for write failure
     */
    public static function writeFailed(string $reason): self
    {
        return new self("Failed to write to SSE stream: {$reason}", 500);
    }
}

File: backend_python/app/exceptions/streaming_exception.py
"""Exception for SSE streaming errors."""
from .ai_assistant_exception import AIAssistantException


class StreamingException(AIAssistantException):
    """Exception for SSE streaming errors."""

    @staticmethod
    def connectionFailed(reason: str):
        """Create exception for connection failure."""
        return StreamingException(f"SSE connection failed: {reason}", 503)

    @staticmethod
    def writeFailed(reason: str):
        """Create exception for write failure."""
        return StreamingException(f"Failed to write to SSE stream: {reason}", 500)
